rpc/provider/smoldot.test.ts

```diff
@@ -147,42 +140,39 @@ Deno.test({
     await t.step({
       name: "invalid chain spec",
       async fn() {
-        const stopped = deferred()
-        const provider = smoldotProvider({ chainSpec: { relay: "" } }, (message) => {
-          assertInstanceOf(message, Error)
-          stopped.resolve()
-        })
-        provider.send({
-          jsonrpc: "2.0",
-          id: provider.nextId(),
-          method: "system_health",
-          params: [false],
-        })
-        await stopped
-        const providerRelease = await provider.release()
-        assertNotInstanceOf(providerRelease, Error)
+        const [ref, message] = await setup(
+          smoldotProvider,
+          { chainSpec: { relay: "" } },
+          "system_health",
+          [false],
+        )
+        assertInstanceOf(message, Error)
+        assertNotInstanceOf(await ref.release(), Error)
       },
     })
     await t.step({
       name: "send non-JSON",
       async fn() {
-        const relay = await (
-          await fetch(
-            "https://raw.githubusercontent.com/paritytech/substrate-connect/main/packages/connect/src/connector/specs/polkadot.json",
-          )
+        const relay = await fetchChainSpec(
+          "https://raw.githubusercontent.com/paritytech/substrate-connect/main/packages/connect/src/connector/specs/polkadot.json",
         )
-          .text()
-        const stopped = deferred()
-        const provider = smoldotProvider({ chainSpec: { relay } }, (message) => {
-          assertInstanceOf(message, Error)
-          stopped.resolve()
-        })
-        // make JSON.stringify to throw
-        provider.send(1n as never)
-        await stopped
-        const providerRelease = await provider.release()
-        assertNotInstanceOf(providerRelease, Error)
+        const [ref, message] = await setup(
+          smoldotProvider,
+          { chainSpec: { relay } },
+          "system_health",
+          // make JSON.stringify to throw
+          [1n],
+        )
+        assertInstanceOf(message, Error)
+        assertNotInstanceOf(await ref.release(), Error)
       },
     })
   },
 })
+
+async function fetchChainSpec(url: string) {
```

Comment:
Seems broadly applicable. Might be good to rename to `fetchText` and place in a new `utils/fetch.ts` file, then import from `utils/mod.ts`?